libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/guidance/sapaguidemap/api/MapboxSapaApi.kt

```diff
@@ -0,0 +1,172 @@
+package com.mapbox.navigation.ui.maps.guidance.sapaguidemap.api
+
+import com.mapbox.api.directions.v5.models.BannerComponents
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.common.ResourceLoadError
+import com.mapbox.common.ResourceLoadResult
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.base.trip.model.roadobject.reststop.RestStop
+import com.mapbox.navigation.ui.base.util.MapboxNavigationConsumer
+import com.mapbox.navigation.ui.maps.guidance.sapaguidemap.SapaAction
+import com.mapbox.navigation.ui.maps.guidance.sapaguidemap.SapaProcessor
+import com.mapbox.navigation.ui.maps.guidance.sapaguidemap.SapaResult
+import com.mapbox.navigation.ui.maps.guidance.sapaguidemap.model.MapboxSapaOptions
+import com.mapbox.navigation.ui.maps.guidance.sapaguidemap.model.SapaMapError
+import com.mapbox.navigation.ui.maps.guidance.sapaguidemap.model.SapaMapValue
+import com.mapbox.navigation.ui.utils.internal.resource.ResourceLoaderFactory
+import com.mapbox.navigation.ui.utils.internal.resource.load
+import com.mapbox.navigation.utils.internal.InternalJobControlFactory
+import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.launch
+
+/**
+ * Mapbox Sapa Api allows you to generate service area and parking area guide maps
+ * for select maneuvers.
+ * @property accessToken String
+ * @property options MapboxSapaOptions
+ */
+class MapboxSapaApi @JvmOverloads constructor(
+    val accessToken: String,
+    val options: MapboxSapaOptions = MapboxSapaOptions.Builder().build()
+) {
+
+    private companion object {
+        private const val ACCESS_TOKEN = "?access_token="
+    }
+    private val mainJobController by lazy { InternalJobControlFactory.createMainScopeJobControl() }
+    private val resourceLoader by lazy { ResourceLoaderFactory.getInstance() }
+
+    /**
+     * The method takes in [BannerInstructions] and generates SA/PA guide map based on the presence of
+     * [BannerComponents] of type [BannerComponents.GUIDANCE_VIEW] and subType [BannerComponents.SAPAGUIDEMAP]
+     * @param instructions object representing [BannerInstructions]
+     * @param consumer informs about the state of sapa guide map.
+     */
+    fun generateSapaGuideMap(
+        instructions: BannerInstructions,
+        consumer: MapboxNavigationConsumer<Expected<SapaMapError, SapaMapValue>>
+    ) {
+        val action = SapaAction.CheckSapaMapAvailability(instructions)
+        when (val result = SapaProcessor.process(action)) {
+            is SapaResult.SapaMapAvailable -> {
+                makeSapaMapRequest(result, consumer)
+            }
+            is SapaResult.SapaMapUnavailable -> {
+                onError(
+                    errorMessage = "No sapa guide map available for current maneuver.",
```

Comment:
```suggestion
                    errorMessage = "No service/parking area guide map available for current maneuver.",
```
I don't think we should freely introduce a "sapa" into our docs and dictionary. Could we be more explicit across the docs?

Reply:
Removed all usage of SAPA from this PR. Let me know.
@LukasPaczos